From 7a400afc4201d377bdb4a695fb0b1615348659cf Mon Sep 17 00:00:00 2001
From: Imagination Technologies <powervr@imgtec.com>
Date: Fri, 20 Jan 2023 16:59:41 +0000
Subject: [PATCH 129/168] zink: add robust_access field to shader key

Since shaders can be shared even between robust and non-robust
contexts, the robustness state needs to be tracked.

Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/20808>
---
 src/gallium/drivers/zink/zink_shader_keys.h | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/src/gallium/drivers/zink/zink_shader_keys.h b/src/gallium/drivers/zink/zink_shader_keys.h
index a068fc018a0..320abcd81d2 100644
--- a/src/gallium/drivers/zink/zink_shader_keys.h
+++ b/src/gallium/drivers/zink/zink_shader_keys.h
@@ -32,7 +32,8 @@ struct zink_vs_key_base {
    bool last_vertex_stage : 1;
    bool clip_halfz : 1;
    bool push_drawid : 1;
-   uint8_t pad : 5;
+   bool robust_access : 1;
+   uint8_t pad : 4;
 };
 
 struct zink_vs_key {
@@ -81,7 +82,8 @@ struct zink_fs_key {
    /* non-optimal bits after this point */
    bool lower_line_stipple : 1;
    bool lower_line_smooth : 1;
-   uint16_t pad2 : 14;
+   bool robust_access : 1;
+   uint16_t pad2 : 13;
 };
 
 struct zink_tcs_key {
@@ -92,7 +94,8 @@ struct zink_tcs_key {
  * ctx->compute_pipeline_state.key.size is set in zink_context_create.
  */
 struct zink_cs_key {
-   uint32_t pad : 32;
+   bool robust_access : 1;
+   uint32_t pad : 31;
 };
 
 struct zink_shader_key_base {
-- 
2.17.1

